src/modules/simulation/gz_plugins/moving_platform_controller/MovingPlatformController.cpp

```diff
@@ -0,0 +1,197 @@
+
+	const gz::math::Vector3d vel_sp_err = _target_vel_sp - _current_vel_sp;
+
+	if (vel_sp_err.Length() < max_accel_mag * dt) {
+		// we are already almost there. set directly to avoid division by 0.
+		_current_vel_sp = _target_vel_sp;
+	} else {
+		// acclerate towards target velocity.
+		const gz::math::Vector3d accel = vel_sp_err.Normalized() * max_accel_mag;
+		_current_vel_sp = _current_vel_sp + accel * dt;
+	}
+
+	updateVelocityCommands(_current_vel_sp);
+
+	sendVelocityCommands();
+}
+
+void MovingPlatformController::updateVelocityCommands(const gz::math::Vector3d &velocity_setpoint)
+{
+	// Velocity and angular velocity = low pass filtered white noise + feedback term.
+
+	gz::math::Vector3d noise_v = gz::math::Vector3d(
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal()
+				     );
+
+	gz::math::Vector3d noise_w = gz::math::Vector3d(
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal()
+				     );
+
+	// Update rates for the filtered white noise.
+	// larger number here = faster movement
+	// write these in terms of time constants?
+	// https://ethz.ch/content/dam/ethz/special-interest/mavt/dynamic-systems-n-control/idsc-dam/Lectures/Signals-and-Systems/Lectures/Lecture%20Notes%209.pdf
+	const double alpha_v = 0.01;
+	const double alpha_w = 0.01;
+
+	// Noise amplitude.
+	// larger number here = bigger movement
+	// we scale it by the update rates to keep total energy constant
+	const double ampl_v = 0.01 / alpha_v;
+	const double ampl_w = 0.01 / alpha_w;
+
+	// For ultra realism we might have axis specific versions of these
+	// constants -- Real ships roll more quickly than they pitch and yaw.
+	// But probably overkill for now.
+
+	_noise_v_lowpass = (1 - alpha_v) * _noise_v_lowpass + alpha_v * noise_v;
+	_noise_w_lowpass = (1 - alpha_w) * _noise_w_lowpass + alpha_w * noise_w;
+
+	_platform_v = ampl_v * _noise_v_lowpass + velocity_setpoint;
+	_platform_w = ampl_w * _noise_w_lowpass;
+
+	// feedback terms to ensure the random walk (= integral of noise)
+	// stays within a realistic region.
+	{
+		// small feedback to maintain height. no attempt to stabilise x and y.
+		const double platform_height = 2.;
+		const gz::math::Vector3d pos_gains(0., 0., 1.);  // [m/s / m]
+		_platform_v += -pos_gains * (_platform_position - gz::math::Vector3d(0., platform_height, 0.));
+
+		// eq. 23 from Nonlinear Quadrocopter Attitude Control (Brescianini, Hehn, D'Andrea)
+		// https://www.research-collection.ethz.ch/handle/20.500.11850/154099
+		const double sgn = _platform_orientation.W() > 0 ? 1. : -1.;
+
+		const double attitude_gain = 1.;
+
+		gz::math::Vector3d q_imag = gz::math::Vector3d(
+						    _platform_orientation.X(),
+						    _platform_orientation.Y(),
+						    _platform_orientation.Z()
+					    );
+
+		_platform_w += -attitude_gain * sgn * q_imag;
```

Comment:
Why are we controlling the platform movement with angular velocity? Can't you just constrain the movement?

Reply:
Any further thoughts on this or should I mark this as resolved?

Reply:
I see now what problem you were trying to solve. However, if you are concerned about the contact dynamics, using body rate commands would still introduce the same problems as you set exact pose, as this imposes a kinematics constraint on the simulation. Therefore, using wrench commands would have been the correct way to solve this.

Reply:
Yes, you're right that technically the same problem with the kinematic constraint persists. However, practically it is now so small as to be negligible. All of our use cases have platform mass >> drone mass, making the kinematic constraint a close approximation of reality. 

Do you consider a second-order, wrench-based implementation a must-have for this PR? (either to account for use cases where drone and platform mass are on similar scales, or as a matter of principle)? Or is your comment more of a suggestion for future improvement?

I'm open to changing it, but from our side it's not a requirement.

Reply:
Okay, makes sense. I will change accordingly early next week, leaving this PR open until then. 

ATM we're not blocked, for testing we just merge this branch with others as necessary :)

Reply:
Adressed this in the newest commits, could you re-review? thanks :)